File: Problems/No.204_Count_Primes/count_primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        
        // parameters
        int prime_cnt = 0;
        std::vector<bool> primes(n, true);

        // start computing (using Sieve of Eratosthenes Method)
        if (n <= 1) return 0;
        for (int i=2; i<n; i++)
            if (primes[i]) {

                prime_cnt++;
                int check_multiple = i;
                while (check_multiple < n) {

                    primes[check_multiple] = false;
                    check_multiple += i;

                }

            }
        
        return prime_cnt;

    }
};

File: Problems/No.204_Count_Primes/count_primes.py
class Solution(object):
    def countPrimes(self, n):
        """
        :type n: int
        :rtype: int
        """

        # parameter
        primes = [True] * n

        # start computing (using Sieve of Eratosthenes Method)
        if (n <= 1):
            return 0

        primes[0] = primes[1] = False
        for i in range(2, int(math.sqrt(n))+1):
            if primes[i]:
                for check_multiple in range(i*i, n, i):
                    primes[check_multiple] = False
        
        return sum(primes)
